app/models/user.rb

```diff
@@ -90,4 +91,21 @@ def authenticate_token_creation
     role
     created_at { |created_at| created_at.strftime("%d %b %y %H:%M") }
   end
+
+  def self.from_omniauth(auth)
+    User.where(provider: auth.provider, uid: auth.uid).first_or_create
+  end
+
+  def self.new_with_session(params, session)
+    super.tap do |user|
+      if (data = session['devise.facebook_data'])
+        info = data['info']
+        name = info['name'].split(' ')
```

Comment:
I think facebook could return first_name, and last_names in separate fields. Check it